script.js

```diff
@@ -1,4 +1,256 @@
+// gamemodes
+var gameMode = "dealing cards";
+
+// global state for some booleans
+
+var cardsFinished = false;
+
+// creating deck automatically using loop
+
+var makeDeck = function () {
+  var deck = [];
+
+  // loop 1: separate to suits hearts spades
+
+  var suitIndex = 0;
+  var suits = ["hearts", "diamonds", "clubs", "spade"];
+  while (suitIndex < suits.length) {
+    var currentSuit = suits[suitIndex];
+    console.log(`current suit: ${currentSuit}`);
+
+    //loop 2: separate to rank 1-13
+
+    var rankCounter = 1;
+    while (rankCounter <= 13) {
+      var cardName = rankCounter;
+      var cardValue = rankCounter;
+
+      // special cases for 1 11 12 13
+
+      if (cardName == 1) {
+        cardName = "ace";
+        cardValue = 1;
+      } else if (cardName == 11) {
+        cardName = "jack";
+        cardValue = 10;
+      } else if (cardName == 12) {
+        cardName = "queen";
+        cardValue = 10;
+      } else if (cardName == 13) {
+        cardName = "king";
+        cardValue = 10;
+      }
+
+      // creating the object
+
+      var cardDeck = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        value: cardValue,
+      };
+
+      console.log(`rank : ${rankCounter}`);
+
+      // creating the object array
+
+      deck.push(cardDeck);
+      rankCounter += 1;
+      console.log(cardDeck);
+    }
+
+    suitIndex += 1;
+  }
+
+  console.log("deck", deck);
+  return deck;
+};
+
+// function to make the deck
+
+var cardDeck = makeDeck();
+
+// shuffling the deck
+
+var getRandomIndex = function (max) {
+  return Math.floor(Math.random() * max);
+};
+
+var shuffleCards = function (cardDeck) {
+  var currentIndex = 0;
+  while (currentIndex < cardDeck.length) {
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    console.log("random card ", randomCard);
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    console.log("random card ", currentCard);
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    console.log("random card after swop", randomCard);
+    console.log("currentcard after swop", currentCard);
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+
+  return cardDeck;
+};
+
+// function to shuffle cards based on the card deck we made
+
+var shuffledDeck = shuffleCards(cardDeck);
+console.log("shuffled deck", shuffledDeck);
+
+// start main function
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  // initialize output
+
+  var output = "";
+
+  //pop takes the card at the end of the array and removes
+
+  var computerCard1 = shuffledDeck.pop();
```

Comment:
consider using a function (containing a loop) to generate player/ computer's initial hands